elements/lisk-tree/src/sparse_merkle_tree/sparse_merkle_tree.ts

```diff
@@ -12,20 +12,30 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-import { EMPTY_HASH } from './constants';
+import { EMPTY_HASH, NodeSide } from './constants';
+import { Leaf } from './leaf';
 import { Database } from './types';
+import { binaryExpansion, bufferToBinaryString, parseBranch, parseLeaf, isLeaf } from './utils';
+import { Branch } from './branch';
+import { Empty } from './empty';
+import { InMemoryDB } from '../inmemory_db';
 
+type TreeNode = Branch | Leaf | Empty;
 export class SparseMerkleTree {
 	private readonly _db: Database;
-	private readonly _rootHash: Buffer;
 	private readonly _keyLength: number;
-	public constructor(db: Database, rootHash?: Buffer, keyLength = 36) {
-		this._db = db;
-		this._keyLength = keyLength;
-		this._rootHash = rootHash ?? EMPTY_HASH;
+	private _rootNode: TreeNode;
+
+	public constructor(options: { db?: Database; rootHash?: Buffer; keyLength?: number }) {
+		this._db = options?.db ?? new InMemoryDB();
+		this._keyLength = options?.keyLength ?? 32;
+		this._rootNode = new Empty();
 	}
 	public get rootHash(): Buffer {
```

Comment:
I think this `get rootHash` is not needed since you can get the `rootNode`. Also, the class does not have any property `rootHash` now.